src/compoents/profile/ProfileCalLinkForm.jsx

```diff
@@ -0,0 +1,92 @@
+import { Button, FormControl, FormLabel, Grid, InputAdornment, OutlinedInput, Typography } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { useAuth } from '../../providers/AuthProvider';
+import { useStatus } from '../../providers/MsgStatusProvider';
+
+export default function ProfileCalLinkForm() {
+    const { user, token } = useAuth();
+    const userDetails = user[0];
+
+    const { setStatusMessage, setIsAlertOpen, setStatusType } = useStatus();
+
+    const [ FormData, setFormData ] = useState({ calendar_link: '' });
+
+    useEffect(() => {
+        if (userDetails) {
+            setFormData({ calendar_link: userDetails?.mentor_details?.calendar_link?.replace(/https:\/\//g, '') || '' });
```

Comment:
I feel that we should use `new URL(userDetails?.mentor_details?.calendar_link)` here so the all URLs are properly parsed, then strip the protocol and rebuild the string.  thoughts?

Reply:
Way cleaner solution! thanks